test/modules/ecs_cluster.go

```diff
@@ -0,0 +1,107 @@
+package modules
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+)
+
+func DeployEcsClusterUsingTerraform(t *testing.T, workingDir string) {
+	// Generate a unique ID
+	uniqueId := random.UniqueId()
+	// Get a random AWS region
+	awsRegion := aws.GetRandomStableRegion(t, []string{"us-east-1", "us-east-2"}, nil)
+	test_structure.SaveString(t, workingDir, "awsRegion", awsRegion)
+	// Get a ECS AMI
+	amiId := aws.GetEcsOptimizedAmazonLinuxAmi(t, awsRegion)
+	// Construct the terraform options with default retryable errors to handle the most common retryable errors in
+	// terraform testing.
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: workingDir,
+		Vars: map[string]interface{}{
+			"random_id":            uniqueId,
+			"region":               awsRegion,
+			"cluster_instance_ami": amiId,
+		},
+	})
+
+	// Save the options so later test stages can use them
+	test_structure.SaveTerraformOptions(t, workingDir, terraformOptions)
+}
+
+func ValidateEcsCluster(t *testing.T, workingDir string) {
```

Comment:
Tests look good. Personally, I think it is a little cluttered together. Is there a way we could add some spaces between the tests. And have a standard comment format to denote the test. Kinda like in TS how we have the "describe ... it" structure.

Reply:
Maybe something like this. Feel free to change it. I just think it might help with organization. This could also give me a format for requirements of test cases.

```go

// CASE: Cluster should exist

// CASE: Cluster status should be active

// CASE: Capacity providers should include FARGATE

// CASE: Default capacity provider should be set

// CASE: Default capacity provider should be custom EC2 ASG

// CASE: At least 1 container instance should be registered in the cluster 
```